### PR TITLE
Remove redundant interfaces from FilterWindow and SendAlertMessageWindow

### PR DESCRIPTION
**Pull request by lusarz**
Interfaces like [these](https://github.com/bisq-network/bisq/blob/d55114e019fda469266dab50f63916289b2e71cb/desktop/src/main/java/bisq/desktop/main/overlays/windows/FilterWindow.java#L60-L69) are redundant, and unnecessarily increase the complexity of the code. In this pull request I remove them and use appropriate methods like `filterManager.addFilterMessageIfKeyIsValid` directly. It's separated into 2 commits:
- [first one](https://github.com/bisq-network/bisq/commit/4dfc69e02cf2ba56634868257863b74e68cd9522) for FilterWindow
- [second one](https://github.com/bisq-network/bisq/commit/5cdb65d19e9702edafa3be0be7297ecae113ca07) for SendAlertMessageWindow

### CODE REVIEW

**Review by freimair — DISMISSED** [review]
utAck. Easier to read for sure!

**Comment by ripcurlx** [comment]
The reason why this is not in the FilterWindow is, that we try not to trigger any business logic directly in a window, but delegate it to the controller/model that triggered the window in the first place.

**Comment by lusarz** [comment]
@ripcurlx Is something wrong with triggering business logic directly in a window ?

**Comment by ripcurlx** [comment]
> @ripcurlx Is something wrong with triggering business logic directly in a window ?

In general no. But we saw them until now as dumb views that just present or request data from the user. So until now in all windows/overlay (as far as I know) there is no business logic directly triggered.

**Comment by lusarz** [comment]
@ripcurlx I see a few places where business logic is triggered directly in a window, for example:
- [SignPaymentAccountsWindow](https://github.com/bisq-network/bisq/blob/df6812a4c958553cfaf4587bf814de9d11be1a87/desktop/src/main/java/bisq/desktop/main/overlays/windows/SignPaymentAccountsWindow.java#L283)
- [DisputeSummaryWindow](https://github.com/bisq-network/bisq/blob/df6812a4c958553cfaf4587bf814de9d11be1a87/desktop/src/main/java/bisq/desktop/main/overlays/windows/DisputeSummaryWindow.java#L645)

And a few candidates for similar refactoring: 
- [UnlockDisputeAgentRegistrationWindow](https://github.com/bisq-network/bisq/blob/f3456f1920bb115a06ece23abdc3136449894895/desktop/src/main/java/bisq/desktop/main/account/register/AgentRegistrationView.java#L124)
- [SendPrivateNotificationWindow here](https://github.com/bisq-network/bisq/blob/e0de9982e2e83713f65e36085e8fc09a24fe9078/desktop/src/main/java/bisq/desktop/main/overlays/editor/PeerInfoWithTagEditor.java#L250) and [here](https://github.com/bisq-network/bisq/blob/e0de9982e2e83713f65e36085e8fc09a24fe9078/desktop/src/main/java/bisq/desktop/main/overlays/editor/PeerInfoWithTagEditor.java#L250)

I think creating unnatural interfaces just to be able to pass single (always same) function from parent is overkill - it only increase complexity with no benefit.

**Comment by ripcurlx** [comment]
> @ripcurlx I see a few places where business logic is triggered directly in a window, for example:
> 
> * [SignPaymentAccountsWindow](https://github.com/bisq-network/bisq/blob/df6812a4c958553cfaf4587bf814de9d11be1a87/desktop/src/main/java/bisq/desktop/main/overlays/windows/SignPaymentAccountsWindow.java#L283)
> * [DisputeSummaryWindow](https://github.com/bisq-network/bisq/blob/df6812a4c958553cfaf4587bf814de9d11be1a87/desktop/src/main/java/bisq/desktop/main/overlays/windows/DisputeSummaryWindow.java#L645)
> 

True - I forgot about those two. I even wrote one of it 😬 

As I also agree that it is more readable the way you refactored it and we've already broken this un-outspoken rule, I'll utACK this PR.

**Review by ripcurlx — DISMISSED** [review]
utACK

**Comment by ripcurlx** [comment]
@lusarz Now you just need to sign your commits to get your PR merged.

**Comment by lusarz** [comment]
@ripcurlx I signed commits - I'll try to do something to have all my commits always signed :)

**Comment by lusarz** [comment]
@ripcurlx , @freimair merge ? It was already approved, but I needed to fix conflicts with master and sign commits.

**Review by ripcurlx — APPROVED** [review]
utACK

**Comment by ripcurlx** [comment]
@lusarz Please don't dismiss existing reviews as they should be resolved by the maintainers themselves. I didn't enforce this on GitHub right now, but it makes it burdensome to do an ACK review again compared to just resolve the required changes on maintainer side.

**Comment by lusarz** [comment]
@ripcurlx I didn't dismissed them intentionally - it happened automatically when I did `git push --force` after signing my commits.

**Comment by ripcurlx** [comment]
> @ripcurlx I didn't dismissed them intentionally - it happened automatically when I did `git push --force` after signing my commits.

🤔 Didn't know that.

**Comment by lusarz** [comment]
@ripcurlx No worry :) This problem is alreasy addressed in github community [here](https://github.community/t5/How-to-use-Git-and-GitHub/Auto-approve-force-pushes-if-PR-was-already-approved/td-p/15029)

**Pull request merged**
